docs/migrations/migrations_guide.md

```diff
@@ -0,0 +1,192 @@
+```
+
+### ✍️ Edit the files:
+
+**Up (apply):**
+
+```sql
+ALTER TABLE users ADD COLUMN nickname TEXT;
+```
+
+**Down (rollback):**
+
+```sql
+ALTER TABLE users DROP COLUMN nickname;
+```
+
+---
+
+## 🕒 When Migrations Run
+
+- Automatically on service startup.
+- On local dev, you can trigger them manually via CLI (see below).
+- Only `.up.sql` files are executed in forward direction, and only if not already applied.
+
+---
+
+## 🔁 Rollbacks
+
+To undo one or more migrations:
+
+### Using CLI:
+
+```bash
+migrate -path ./migrations -database <DB_URL> down 1     # Rollback last 1 migration
+migrate -path ./migrations -database <DB_URL> down        # Rollback all
+```
+
+### Programmatically:
+
+In Go:
+
+```go
+_ = m.Steps(-1)   // Rollback one
+_ = m.Down()      // Rollback all
+```
+
+### Dirty state recovery:
+
+If a migration fails midway:
+- The DB may be marked “dirty”
+- Use `migrate force <version>` to fix state before rerunning
+
+---
+
+## 🧪 How to Test Migrations
+
+1. Spin up a local DB (Docker or Postgres locally)
+2. Run all migrations up:
+   ```bash
+   migrate -path ./migrations -database <DB_URL> up
+   ```
+3. Run all down:
+   ```bash
+   migrate -path ./migrations -database <DB_URL> down
+   ```
+4. Run tests (if defined) to verify schema is as expected
+
+---
+
+## 🧾 Naming Conventions
+
+- Use `snake_case` for clarity
+- Be descriptive: `add_index_to_org_table`, `create_user_settings_table`, etc.
+- Keep versions sequential (`000001`, `000002`, ...) or use timestamps (`20250615120000_create_table.up.sql`)
+
+---
+
+## 🛠️ Best Practices
+
+- Keep migrations **small and focused**
+- Always add both `.up.sql` and `.down.sql`
+- **Test migrations** in a clean DB locally before pushing
+- Do **not edit** already-applied migration files—create a new one
+- Include migration tests in CI/CD
+- Prefer **embedded migrations** in binary for consistent deploys
+
+---
+
+## 📎 Helpful Commands
+
+```bash
+# Install CLI
+brew install golang-migrate
+
+# Run migrations
+migrate -path ./migrations -database postgres://user:pass@localhost:5432/db up
+
+# Rollback
+migrate -path ./migrations -database postgres://user:pass@localhost:5432/db down 1
+
+# Fix dirty state
+migrate -path ./migrations -database <DB_URL> force <version>
+```
+
+---
+
+## 📌 Example Migration
+
+### `000003_add_status_to_orders.up.sql`
+
+```sql
+ALTER TABLE orders ADD COLUMN status VARCHAR(20) DEFAULT 'pending';
+```
+
+### `000003_add_status_to_orders.down.sql`
+
+```sql
+ALTER TABLE orders DROP COLUMN status;
+```
+
+---
+
+## 🙋 Common Issues
+
+| Issue | Fix |
+|------|-----|
+| Migration failed and DB is "dirty" | Run `migrate force <version>` |
+| "No change" message | All migrations already applied |
+| Rollback fails | Ensure `.down.sql` is defined and reversible |
+| Duplicate version error | Check your filenames and sequencing |
+
+---
+
+## 📫 Questions?
+
+Reach out in the #backend Slack channel or ping `@<migration-maintainer>` if you're stuck!
```

Comment:
We do not have any backend slack channel, you can provide Discord Channel link and raise query in #nixopus-support channel or create thread in #question-helps-bugs